contribute/e2e/CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard 
+application images from dockerhub as well as custom-built images stored in openebs/test-storage).Since the tests are performed 
+in a kubernetes environment, the test images are typically deployed as "pods" and "jobs" and therefore accompanied by the 
+respective specification YAML files.
+
+## BUILDING BLOCKS OF OPENEBS CI
+
+The OpenEBS Continuous Integration has two distinct parts : 
+
+- Travis is used to run unit tests, build code and push images to dockerhub (in case of repos where the image is the build artifact)
+- Jenkins pulls these images, creates the kubernetes-based test environment and executes the e2e. It is also used to run tests upon commits that change application specs OR openebs kubernetes deployment specs.
+
+While the OpenEBS e2e can be run as a standalone suite by users, the priority is have them run upon commits to the core openebs github
+repositories such as openebs/openebs, openebs/mayaserver, openebs/jiva. This part of the continuous integration is built around a dedicated Jenkins server, which upon detecting commits via a polling mechanism, launches VMs using Vagrant and runs the ansible playbooks to setup the environment and execute e2e tests. The openebs/e2e project focuses on this part of the CI.
+
+The below schema illustrates the workflow
+
+![OpenEBS CI Overview Diagram](https://github.com/ksatchit/openebs/blob/master/documentation/source/_static/OpenEBS_CI_Workflow.png)
+
+## CONTRIBUTING TO E2E & OPENEBS CI 
+
+Contributions are welcome !!  Before starting, a reading of the project documentation at [ReadTheDocs](http://openebs.readthedocs.io/en/latest/ ) is highly recommended. Please raise a github issue for any questions you may have on the project in general or e2e,CI in particular.
+You can post questions on the OpenEBS slack channel too.
+
+You could contribute to any of the following areas : 
+
+- Submit proposals to add new tests/test categories to the e2e
+- Create new application specification YAMLs 
+- Create ansible playbooks to execute test workflows
+- Submit changes to fix bugs/improve e2e tests 
+- Raise/fix issues on the CI framework 
+- Raise/fix issues on the e2e, CI documentation
+
+## GENERAL CONTRIBUTION GUIDELINES
+
+Here are some general contribution guidelines. 
+
+### Adding new tests/test-categories
+
+To add new tests, please raise an issue describing the objective, pre-requisites, test steps, expected results & cleanup routines.
+In case of proposals to add new test sections/categories, also add a generic description explaining the purpose and value addition
+
+Once accepted, the issue can be assigned to interested contributors who can make relevant changes with a pull request. 
+
+### Create new application YAMLs
+
+To add new applications to e2e, please consider the points made here, as applicable : 

```

Comment:
please consider the following points, if applicable:

Reply:
Done